core/chains/evm/logpoller/orm.go

```diff
@@ -37,6 +37,53 @@ func (o *ORM) InsertBlock(h common.Hash, n int64, qopts ...pg.QOpt) error {
 	return err
 }
 
+// InsertFilter is idempotent.
+//
+// Each address/event pair must have a unique job id, so it may be removed when the job is deleted.
+// If a second job tries to overwrite the same pair, this should fail.
+func (o *ORM) InsertFilter(filter Filter, qopts ...pg.QOpt) (err error) {
+	q := o.q.WithOpts(qopts...)
+	addresses := make([][]byte, 0)
+	events := make([][]byte, 0)
+
+	for _, addr := range filter.Addresses {
+		addresses = append(addresses, addr.Bytes())
+	}
+	for _, ev := range filter.EventSigs {
+		events = append(events, ev.Bytes())
+	}
+	err = q.ExecQ(`INSERT INTO log_poller_filters
+								(filter_name, evm_chain_id, created_at, address, event)
+								SELECT * FROM
+									(SELECT $1, $2::NUMERIC, NOW()) x,
+									(SELECT unnest($3::BYTEA[]) addr) a,
+									(SELECT unnest($4::BYTEA[]) ev) e
+								ON CONFLICT (filter_name, evm_chain_id, address, event) DO NOTHING`,
+		filter.FilterName, utils.NewBig(o.chainID), addresses, events)
+	return err
+}
+
+// DeleteFilter removes all events,address pairs associated with the filter
+func (o *ORM) DeleteFilter(name string, qopts ...pg.QOpt) error {
+	q := o.q.WithOpts(qopts...)
+	err := q.ExecQ(`DELETE FROM log_poller_filters WHERE filter_name = $1`, name)
+	return err
+}
+
+// LoadFiltersForChain returns all filters for this chain
+func (o *ORM) LoadFilters(qopts ...pg.QOpt) (map[string]Filter, error) {
+	q := o.q.WithOpts(qopts...)
+	rows := make([]Filter, 0)
+	err := q.Select(&rows, `SELECT filter_name, ARRAY_AGG(DISTINCT address)::BYTEA[] AS addresses, ARRAY_AGG(DISTINCT event)::BYTEA[] AS event_sigs
```

Comment:
Yes, although it's not very specific to this query.  We will need them any time we want to read data directly into a `Filter` struct, or any other struct which includes arrays of `common.Address` or `common.Hash`.  The underlying reason is because `pgx` doesn't know how to automatically convert from `[]BYTEA[]` to `[20][]byte`, it only works with a data type that looks like `[][]byte`.  Fixed length types and variable length types are handled very differently in sql, and unfortunately there is no fixed length type that looks like `BYTEA[20]`, there is only `BYTEA[]`.  We enforce the 20 with a constraint, but it's not constraint-aware enough to understand that.